pkg/component/istio/test_charts/ingress_autoscaler.yaml

```diff
@@ -20,4 +20,23 @@ spec:
       name: cpu
       target:
         averageUtilization: 80
-        type: Utilization
\ No newline at end of file
+        type: Utilization
+  - type: Resource
+    resource:
+      name: memory
+      target:
+        averageUtilization: 80
+        type: Utilization
+  behavior:
+    scaleDown:
+      stabilizationWindowSeconds: 1800
+      policies:
+      - type: Pods
+        value: 1
+        periodSeconds: 1800
+    scaleUp:
+      stabilizationWindowSeconds: 60
+      policies:
+      - type: Pods
+        value: 1
+        periodSeconds: 60
```

Comment:
Without looking at historic data, I would not make the scale up slower as it is today and  keep the default for scaling up. When I understand the docs correctly, this would be 4 pods and 15 sec.

Reply:
Do you mind referencing the docs? I did not find the defaults for scaling up.

Reply:
From what I understand, https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/autoscaling/v2/defaults.go might suggest that 4 pods and 15 seconds might be the defaults. However, according to https://github.com/kubernetes/kubernetes/blob/d171e6733a72bd9fce2a6898c6de1f3bb2ff0e5e/pkg/apis/autoscaling/v2/defaults.go#L97 these defaults are only used if `.spec.behavior` was specified, which we did not in the past. In the actual implementation, I failed to find such defaulting logic (see https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/podautoscaler/horizontal.go), but there are also fairly different paths for with or without `.spec.behavior`.

WDYT?

Reply:
Without having `.spec.behavior` specified, we scale-up instantly to the desired number of pods:

``
desiredReplicas = ceil[currentReplicas * ( currentMetricValue / desiredMetricValue )]
``

The scale down would happen after the time specified by `--horizontal-pod-autoscaler-downscale-stabilization` . This is a configuration option of the kube-controller-manager. For our seeds we configure `24h` here.
So, our current default is  faster scale-up and much slower at scale-down compared to this approach.

IMHO, we could start without the `spec.behavior` configuration, as it's the more conservative approach.
In case we see, that small CPU spikes keep the number of replicas up all the time, we can add a suited hpa config with that knowledge.

Reply:
An advantage of an explicit specification of the scaling policy, is that you don't need to dig into KCM configuration to get the actual settings.
The configured scale-up should be fast enough.